editor/components/inserter/menu.js

```diff
@@ -95,7 +97,14 @@ export class InserterMenu extends Component {
 	}
 
 	isDisabledBlock( blockType ) {
-		return blockType.useOnce && find( this.props.blocks, ( { name } ) => blockType.name === name );
+		const requiredCapability = get( blockType, 'capability' );
+		let isDisabled = blockType.useOnce && find( this.props.blocks, ( { name } ) => blockType.name === name );
+
+		if ( ! isDisabled && requiredCapability ) {
+			isDisabled = ! get( this.props.user.data, [ 'capabilities', requiredCapability ] );
+		}
```

Comment:
Heads up: this logic has moved to `selectors.getInserterItems` (see https://github.com/WordPress/gutenberg/pull/4497).